### PR TITLE
pass through KUBE_BUILD_PLATFORMS to enable build on specified platforms

### PR DESCRIPTION
**Pull request by dixudx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Currently `make cross` will build targets for all platforms. And `KUBE_FASTBUILD` will build for linux/amd64.

But what if I only want a single build for a specified platform, such as `linux/arm64`. Using `make cross` could help build `linux/arm64` for me, but other platforms will be built as well. Using `KUBE_BUILD_PLATFORMS=linux/arm64 make release` does not help, since this env is not passed down to the building container.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by spiffxp** [comment]
/release-note-none
nit: I don't think this is a user-facing change, please update the PR description

I'm confused what this is fixing, is this not the path by which we cross-build today?

File: build/common.sh [hunk 1]
```diff
@@ -569,6 +569,7 @@ function kube::build::run_build_command_ex() {
     --env "KUBE_BUILDER_OS=${OSTYPE:-notdetected}"
     --env "KUBE_VERBOSE=${KUBE_VERBOSE}"
     --env "KUBE_BUILD_WITH_COVERAGE=${KUBE_BUILD_WITH_COVERAGE:-}"
+    --env "KUBE_BUILD_PLATFORMS=${KUBE_BUILD_PLATFORMS:-}"
```
Comment (BenTheElder) [inline build/common.sh]: cc @mkumatag

**Comment by dixudx** [comment]
@spiffxp Currently `make cross` will build targets for all platforms. And `KUBE_FASTBUILD` will build for `linux/amd64`.

But what if I only want a single build for a specified platform, such as `linux/arm64`. Using `make cross` could help build `linux/arm64` for me, but other platforms will be built as well. Using `KUBE_BUILD_PLATFORMS=linux/arm64 make release` does not help, since this env is not passed down to the building container.

**Comment by spiffxp** [comment]
OK that makes sense to me
/retest
/lgtm
I'll /approve once you've updated PR description to drop the release note

**Comment by BenTheElder** [comment]
ping
/retest

**Comment by dixudx** [comment]
@spiffxp Already dropped the release note.

**Comment by cblecker** [comment]
/approve
/priority backlog
/sig release
FYI @kubernetes/release-engineering

**Comment by mkumatag** [comment]
I see an error while running on the x86 machine:

**Pull request merged**